src/api/setupMockWorker.ts

```diff
@@ -1,17 +1,29 @@
-import { setupWorker, MockedRequest, RestHandler } from 'msw'
+import { setupWorker, MockedRequest, RestHandler, SetupWorker } from 'msw'
+
+let mockWorker: SetupWorker | undefined
+
+/**
+ * Keeps track of handlers per API.
+ */
+const handlersMap = new Map<string, RestHandler[]>()
 
 /**
  * Sets up the msw-based mock server.
  */
-export function setupMockWorker(handlers: RestHandler[]): void {
-  const worker = setupWorker(...handlers)
+export function setupMockWorker(apiName: string, handlers: RestHandler[] = []) {
+  handlersMap.set(apiName, handlers)
+
+  const allHandlers = Array.from(handlersMap.values()).flat()
+  // Stops an existing service worker so new listeners are being registered. The warning this causes in the dev tools console does **NOT** seem to be accurate.
+  mockWorker?.stop()
```

Comment:
**Comment**: This cost me several hours of development work. When adding it, MSW logs a message that calling stop for a running worker has no effect and that it should be removed. Only much later did I realize that this isn’t correct.